Lib/platform.py

```diff
@@ -1464,9 +1464,24 @@ def invalidate_caches():
 
 ### Command line interface
 
-if __name__ == '__main__':
-    # Default is to print the aliased verbose platform string
-    terse = ('terse' in sys.argv or '--terse' in sys.argv)
-    aliased = (not 'nonaliased' in sys.argv and not '--nonaliased' in sys.argv)
-    print(platform(aliased, terse))
-    sys.exit(0)
+def _parse_args(args: list[str] | None):
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--terse", action="store_true")
```

Comment:
I am not quite sure about `not 'nonaliased' in sys.argv` part. Basically now people can pass 

```
» ./python.exe -m platform    
macOS-14.5-arm64-arm-64bit-Mach-O
                                                                                                                                                        
» ./python.exe -m platform terse
macOS-14.5
```

Now this behavior was changed.

Reply:
Thanks, I'm not sure how this slipped through the cracks when looking at this :). I have an idea about fixing this, but I'm not sure its the best implementation.

Its a slightly unusual interface to implement from argparse's perspective, since argparse doesn't appear to support flags without a prefix.

Reply:
argparse now accepts `terse` and `nonaliased` as optional positional arguments, this still breaks some of the really obscure edge cases e.g.

```
./python.exe -m platform terse --terse terse
```

but I think it should work for the sane use-cases